integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/correctness/runner/resultset/Row.java

```diff
@@ -56,4 +57,32 @@ private Object roundFloatNum(Object value) {
     return value;
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  public int compareTo(Row other) {
+    List<Object> thisObjects = new ArrayList<>(values);
+    List<Object> otherObjects = new ArrayList<>(other.values);
+
+    for (int i = 0; i < thisObjects.size(); i++) {
+      Object thisObject = thisObjects.get(i);
+      Object otherObject = otherObjects.get(i);
+
+      /*
+       * Only one is null, otherwise (both null or non-null) go ahead.
+       * Always consider NULL is greater which means NULL comes last in ASC and first in DESC
+       */
+      if (thisObject == null ^ otherObject == null) {
+        return thisObject == null ? 1 : -1;
+      }
```

Comment:
> Shall we also include missing values in compareTo?

> Shall we also include missing values in compareTo?

Sure, the object here is what we get from JSON response. So I think no matter missing value absent or present as ex."MISSING", it will be taken care by the `if` below. Let me know if any case I miss. Thanks.